server.js

```diff
@@ -25,11 +41,67 @@ const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json());
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// endpoint to get all books
+app.get('/allBooks', async (req, res) => {
+  const books = await Book.find()
+  res.json(books)
+})
+
+// endpoint to get a single book by id
+app.get('/id/:id', async (req, res) => {
+  if (!mongoose.isValidObjectId(req.params.id)) {
+    res.status(404).json({ error: 'Invalid object id syntax' })
+    return;
+  }
+
+  try {
+    const book = await Book.findById(req.params.id)
+    if (book) {
+      res.json(book)
+    } else {
+      res.status(404).json({ error: 'Book not found' })
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Something went wrong', details: error })
+  }
+})
+
+app.get('/textreviews', async (req, res) => {
+  let books = await Book.find()
+  books = books.sort((a, b) => b.text_reviews_count - a.text_reviews_count);
+  res.json(books.slice(0, 20));
```

Comment:
here as well you can limit sent data by using mongoose functions, not operate on the already downloaded to the server data.